### PR TITLE
LPD-26931

### PR DESCRIPTION
**Pull request by ChrisKian**
https://liferay.atlassian.net/browse/LPD-26931

ResourcePermission needs to be included for each new SAP entry.  Will also need to be backported to 2024.Q2

### CODE REVIEW

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
None of the failing upgrade tests are the modified file `SAPEntryUpgradeProcess.java`.  PR should be good for forwarding

**Comment by ChrisKian** [comment]
ci:test:sf

File: modules/apps/portal-security/portal-security-service-access-policy-service/src/main/java/com/liferay/portal/security/service/access/policy/internal/upgrade/v3_0_1/SAPEntryUpgradeProcess.java [hunk 1]
```diff
@@ -95,6 +103,10 @@ protected void doUpgrade() throws Exception {
 								companyId)));
 
 					preparedStatement2.execute();
+
+					ResourceLocalServiceUtil.addResources(
+						companyId, 0, guestUserId, SAPEntry.class.getName(),
+						sapEntryId, false, false, false);
```
Comment (stian-sigvartsen) [inline modules/apps/portal-security/portal-security-service-access-policy-service/src/main/java/com/liferay/portal/security/service/access/policy/internal/upgrade/v3_0_1/SAPEntryUpgradeProcess.java:109]: Hi @ChrisKian . Curious why you decided to call `ResourcLocalServiceUtil#addResources` here and in `SAPEntryLocalServiceImpl#checkSystemSAPEntries` you called `ResourcePermissionLocalService#setResourcePermissions` for the same purpose?

Comment (ChrisKian) [reply modules/apps/portal-security/portal-security-service-access-policy-service/src/main/java/com/liferay/portal/security/service/access/policy/internal/upgrade/v3_0_1/SAPEntryUpgradeProcess.java:109]: I just used the code from Pedro's script, as it seems to work fine.

I will adjust this code to follow the checkSystemSAPEntries logic more closely.  Thanks for catching this

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
ci:test:upgrade

**Comment by ChrisKian** [comment]
None of the upgrade failures are from the `SAPEntryUpgradeProcess` class.

**Comment by stian-sigvartsen** [comment]
Looks good @ChrisKian !

**Comment by stian-sigvartsen** [comment]
ci:forward

**Pull request closed without merging**